### PR TITLE
Ability to subscribe/unsubscribe down to the event type

### PR DESCRIPTION
**Pull request by fhlavac**
[RHCLOUD-21306](https://issues.redhat.com/browse/RHCLOUD-21306)

- enabled new event-based notification preferences
- cleaned the code
- fixed BulkSelectButton state issues
- enhanced tests

Before:
image

Now:
image

### CODE REVIEW

File: src/PresentationalComponents/Notifications/utils.js [hunk 1]
```diff
@@ -1,66 +1,104 @@
...
-                ...(emailSection[0] === section.name &&
-                emailConfig[emailSection[0]]?.bundle === key
-                  ? emailSection?.[1]?.schema?.[0]?.fields?.map((field) => {
-                      unsubscribeAllInactive =
-                        unsubscribeAllInactive && field.initialValue;
+        title: bundleData?.label,
+        name: bundleKey,
+        fields: Object.entries(bundleData.applications).reduce(
+          (acc, [appKey, appData]) => {
+            let selectAllActive = true;
```
Comment (karelhala) [inline src/PresentationalComponents/Notifications/utils.js:17]: This is fairly complicated, we are probably out of luck with something less complex, right?

File: src/api.js [hunk 2]
```diff
@@ -23,14 +23,15 @@ export const getApplicationSchema = (
...
-export const saveValues = async (
-  application,
-  values,
-  apiVersion = 'v1',
-  resourceType = '',
-  url
-) =>
+export const getNotificationsSchema = (apiVersion = 'v1') =>
+  instance.get(
+    `/api/notifications/${apiVersion}/user-config/notification-event-type-preference`
```
Comment (karelhala) [inline src/api.js:28]: We should update the javascript clients to include notifications client as well.

**Review by karelhala — APPROVED** [review]
Looking good! Just one question wrt complexity of calculation, but I don't think we can do anything interesting with it.

File: src/PresentationalComponents/Notifications/utils.js (hunk 1 above)
Comment (fhlavac) [reply src/PresentationalComponents/Notifications/utils.js:17]: this is the best I've been able to come up with :laughing: 
If we don't want the backend to add more UI details to the schema, we need to loop through it and wrap it with the remaining components.

File: src/api.js (hunk 2 above)
Comment (fhlavac) [reply src/api.js:28]: https://issues.redhat.com/browse/RHCLOUD-27927

**Review by karelhala — APPROVED** [review]

**Pull request merged**